### PR TITLE
Stats

### PR DESCRIPTION
**Pull request by Kirill888**
bunch of stuff

### CODE REVIEW

File: libs/stats/odc/stats/_pq.py [hunk 1]
```diff
@@ -42,40 +52,89 @@ def pq_product(location: Optional[str] = None) -> OutputProduct:
...
+    .valid            -> .total
+    .erased           -> .clear
+    .erased{postfix}  -> .clear{postfix} For All {postfix}
     """
-    pq = xr.Dataset(dict(clear=xx.clear.sum(axis=0, dtype='uint16'),
-                         total=xx.valid.sum(axis=0, dtype='uint16')))
+    valid = xx.valid
+    erased_bands = [str(n) for n in xx.data_vars if str(n).startswith("erased")]
+    total = valid.sum(axis=0, dtype='uint16')
+    pq = xr.Dataset(dict(total=total))
```
Comment (tebadi) [inline libs/stats/odc/stats/_pq.py:122]: Shouldn't this just be Dict(total)?

File: libs/algo/odc/algo/_masking.py [hunk 2]
```diff
@@ -458,17 +579,87 @@ def _first_valid_np(*aa: np.ndarray, nodata: Union[float, int, None] = None) ->
...
+
+
+def _fuse_and_np(*aa: np.ndarray) -> np.ndarray:
+    """
+    Element wise bit and
+    """
+    assert len(aa) > 0
+    out = aa[0].copy()
+    for a in aa[1:]:
+        out &= a
```
Comment (tebadi) [inline libs/algo/odc/algo/_masking.py:618]: Not an issue, but can rely on numpy.bitwise_and.

File: libs/stats/odc/stats/_pq.py [hunk 3]
```diff
@@ -1,34 +1,44 @@
...
 
 
-bad_pixel_classes = (0, 'saturated or defective')
 cloud_classes = ('cloud shadows',
                  'cloud medium probability',
                  'cloud high probability',
                  'thin cirrus')
 
+default_filters = [(2, 5),
+                   (0, 5)]
```
Comment (tebadi) [inline libs/stats/odc/stats/_pq.py:29]: Please document these number.

**Review by tebadi — APPROVED** [review]

File: libs/algo/odc/algo/_masking.py (hunk 2 above)
Comment (Kirill888) [reply libs/algo/odc/algo/_masking.py:618]: I'm not sure if the equivalent function based implementation is much clearer than an operator based one

```python
    out = aa[0].copy()
    for a in aa[1:]:
        np.bitwise_and(out, a, out=out)
```

**Pull request merged**